src/main/java/bio/terra/service/snapshotbuilder/SnapshotRequestDao.java

```diff
@@ -89,6 +93,24 @@ public SnapshotAccessRequestModel getById(UUID requestId) {
     }
   }
 
+  /**
+   * Get the Snapshot Request associated with the given created snapshot id.
+   *
+   * @param createdSnapshotId associated with one snapshot request.
+   * @return the specified snapshot request or exception if it does not exist.
+   */
+  public SnapshotAccessRequestModel getByCreatedSnapshotId(UUID createdSnapshotId) {
+    String sql = "SELECT * FROM snapshot_request WHERE created_snapshot_id = :created_snapshot_id";
```

Comment:
Do you want to exclude `DELETED` here too? That's what other `get()` methods do.

Reply:
Hm the getById for SnapshotRequestDao doesn't do this. Maybe I should update both methods?

Reply:
I think actually we do want to return deleted requests for the getByCreatedId call. Because when a source snapshot is deleted, all of the snapshot access requests on that snapshot are deleted. However, created snapshots from those requests will still exist as will the sam groups associated with them, stored on the snapshot access request. So, when one of those created snapshots is deleted, we still want to be able to find the request that created it (even if it has been deleted) so that we can delete the sam group stored on it. Also, for the getById endpoint I think we might also want to return deleted requests as well for auditing purposes, but not as sure on this one.

Reply:
> the getById for SnapshotRequestDao doesn't do this

Yes, that's a case where we can't exclude deleted as the caller may want to see a deleted record.

> when one of those created snapshots is deleted, we still want to be able to find the request that created it (even if it has been deleted) so that we can delete the sam group stored on it

OK, that seems reasonable to me. Since this is an internal API (vs a web API) it's OK to return deleted elements. I wonder if the filtering of deleted items for the API call should be done outside the DAO layer and in the service or API layer.